src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using Azure.Monitor.OpenTelemetry.LiveMetrics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Abstractions;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using static Microsoft.AspNetCore.Hosting.Internal.HostingApplication;
+using static Microsoft.Azure.AppService.Proxy.Runtime.Configuration.PolicyValidationError;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics.OpenTelemetry
+{
+    internal static class OpenTelemetryConfigurationExtensions
+    {
+        internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder, HostBuilderContext context)
+        {
+            loggingBuilder
+                .AddOpenTelemetry(o =>
+                {
+                    o.SetResourceBuilder(ConfigureResource(ResourceBuilder.CreateDefault()));
+                    o.AddOtlpExporter();
+                    o.AddAzureMonitorLogExporter();
+                    o.IncludeFormattedMessage = true;
+                    o.IncludeScopes = false;
+                })
+                // These are messages piped back to the host from the worker - we don't handle these anymore if the worker has OpenTelemetry enabled.
+                // Instead, we expect the user's own code to be logging these where they want them to go.
+                .AddFilter<OpenTelemetryLoggerProvider>("Host.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Function.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Azure.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Microsoft.Azure.WebJobs.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled);
+
+            loggingBuilder.Services.AddOpenTelemetry()
+                .ConfigureResource(r => ConfigureResource(r))
+                .WithTracing(b => b
+                    .AddSource("Azure.*")
+                    .AddAspNetCoreInstrumentation()
+                    .AddHttpClientInstrumentation(o =>
+                    {
+                        o.FilterHttpRequestMessage = _ =>
+                        {
+                            Activity activity = Activity.Current?.Parent;
+                            return (activity == null || !activity.Source.Name.Equals("Azure.Core.Http")) ? true : false;
+                        };
+                    })
+                    .AddLiveMetrics()
+                    .AddAzureMonitorTraceExporter()
```

Comment:
Are we going to support Geneva Trace Exporter? Or allow custom exporters to be opt in here for logs, metrics, and traces? Because I believe a lot of Microsoft internal teams would heavily depends on Geneva Exporter to send telemetry to Geneva first then consumed by Jarvis/Azure Managed Grafana.

Reply:
Yes, our plan is to add support for Geneva Exporter in the next release.

Reply:
Thanks. That means a lot to us!